integration/heapster_api_test.go

```diff
@@ -190,6 +207,31 @@ func getSchema(fm kubeFramework, svc *kube_api.Service) (*api_v1.TimeseriesSchem
 	return &timeseriesSchema, nil
 }
 
+func getModelMetrics(fm kubeFramework, svc *kube_api.Service, pod *kube_api.Pod) (*api_v1.MetricResultList, error) {
+	url := fmt.Sprintf("/api/v1/model/namespaces/%s/pod-list/%s/metrics/%s",
```

Comment:
Would unit-like tests validate the restful layer as well? If not, at-least
poking those endpoints to make sure they work will be useful.

On Tue, Sep 15, 2015 at 2:42 PM, Marcin Wielgus <notifications@github.com>
wrote:

> In integration/heapster_api_test.go
> <https://github.com/kubernetes/heapster/pull/558#discussion_r39569407>:
>
> > @@ -190,6 +207,31 @@ func getSchema(fm kubeFramework, svc *kube_api.Service) (*api_v1.TimeseriesSchem
> >  	return &timeseriesSchema, nil
> >  }
> >
> > +func getModelMetrics(fm kubeFramework, svc *kube_api.Service, pod *kube_api.Pod) (*api_v1.MetricResultList, error) {
> > +	url := fmt.Sprintf("/api/v1/model/namespaces/%s/pod-list/%s/metrics/%s",
>
> I would rather check the other endpoints in a more mocked and unit-like
> test for api/v1/model_handler.go where I can verify if the returned
> values are correct.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/heapster/pull/558/files#r39569407>.
>